src/net_processing.cpp

```diff
@@ -1130,9 +1307,16 @@ static bool BlockRequestAllowed(const CBlockIndex* pindex, const Consensus::Para
         (GetBlockProofEquivalentTime(*pindexBestHeader, *pindex, *pindexBestHeader, consensusParams) < STALE_RELAY_AGE_LIMIT);
 }
 
-PeerManager::PeerManager(const CChainParams& chainparams, CConnman& connman, BanMan* banman,
-                         CScheduler& scheduler, ChainstateManager& chainman, CTxMemPool& pool,
-                         bool ignore_incoming_txs)
+std::unique_ptr<PeerManager> PeerManager::make(const CChainParams& chainparams, CConnman& connman, BanMan* banman,
```

Comment:
Adding a new parameter to the ctor now requires updating five locations (`PeerManager::make` declaration, `PeerManager::make` definition signature, `PeerManager::make` function body` `PeerManagerImpl` ctor declaration, `PeerManagerImpl` ctor definition signature.

I wonder if it makes sense to create a `PeerManagerArgs` struct that's passed into the `make()` method and forwarded to the `PeerManagerImpl` ctor so that only one place needs to be changed?

Reply:
Also requires touching everywhere that calls the ctor. Not against it, but doesn't seem like a big win.

Reply:
Not necessarily if we're passing in arguments with defaults. Anyway, not important here. Can be considered separately.

Reply:
BTW, as part of turning (almost) all of net_processing.cpp into `PeerManagerImpl` methods, we could plausibly remove all the `PeerManagerImpl` declarations by just having net_processing.h start with `namespace { class PeerManagerImpl : public PeerManager {` and end with the closing brackets and the `PeerManager::make` definition -- there's no need to declare class methods before use. That'd remove one of the redundancies above, as well as a bunch of others.

Reply:
Seems reasonable, although like you say earlier, it's not a big win. Perhaps something for a follow-up once more of #20758 has landed.